### ISSUE TITLE
[java] ConstructorCallsOverridableMethod ignores abstract methods

### ISSUE THREAD

**Issue by oowekyala** [pmd/pmd#3351]
**Affects PMD Version:** 6.35.0

**Rule:**

Please provide the rule name and a link to the rule documentation:
<https://pmd.github.io/latest/pmd_rules_java_errorprone.html#constructorcallsoverridablemethod>

**Description:** The rule ignores abstract methods:

https://github.com/pmd/pmd/blob/b1e7bc1c9c4db681e4a0f5c9b5bc072fa6d82108/pmd-java/src/test/resources/net/sourceforge/pmd/lang/java/rule/errorprone/xml/ConstructorCallsOverridableMethod.xml#L133-L134

This looks contrary to what the rule is supposed to do.

**Code Sample demonstrating the issue:**

```java
public class Foo {
    public Foo() {
        bar();
    }
    abstract void bar();
}
```

**Expected outcome:**

PMD should report a violation but doesn't. This is a false-negative.

**Running PMD through:** any

**Comment by adangel** [comment]
This has been fixed with PMD 7.0.0-rc1.
